### ISSUE TITLE
EmailComponent throwing Fatal error on missing or invalid subject

### ISSUE THREAD

**Issue by nebojsac** [cakephp/cakephp#7176]
When sending an email with either a missing subject or a subject string with weird utf8 characters, EmailComponent throws a Fatal error on this line:

https://github.com/cakephp/cakephp/blob/2.8/lib/Cake/Controller/Component/EmailComponent.php#L312

Is there a reason for why $this->subject isn't being checked like the other parts of the email?

I'm not sure if it should just return false or throw an Exception(and which exception it should throw), but I guess something like this would be safer:

```php
if (is_object($lib->subject($this->subject))) {
	$lib->subject($this->subject)->messageID($this->messageId);
} else {
	throw new CakeException(__d('cake_dev', 'Invalid subject string', $this->subject));
}
```

I mean, it can't be worse than the uncatchable Fatal error it's throwing now. It's particularly problematic with long running send queue crons.

**Comment by markstory** [comment]
I think checking and raising an exception is a reasonable approach.

**Comment by markstory** [comment]
Fixed with #7179
